### PR TITLE
feat(AI-proxy-plugin): support the new /v1/messages API provided by Anthropic

### PR DESCRIPTION
**Pull request by liverpool8056**
### Summary

Support the newly provided API `/v1/messages` of Anthropic in AI-proxy plugin.
In this PR, the upstream path of Anthropic for `llm/v1/chat` route type is changed from pointing to `/v1/complete` to now pointing to the new API path `/v1/messages`.

A doc page PR is needed correspondingly for this update.

### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been added to `CHANGELOG/unreleased/kong` or adding `skip-changelog` label on PR if unnecessary. [README.md](https://github.com/Kong/kong/blob/master/CHANGELOG/README.md)
- [ ] The Pull Request has backports to all the versions it needs to cover
- [x] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

FTI-5770

### CODE REVIEW

**Comment by ttyS0e** [comment]
I think you have got this bang on. I think you're right that we don't need to support both old and new Claude.

File: changelog/unreleased/kong-ee/add-messages-api-to-anthropic.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+"message": "**AI-Proxy**: To support the new messages API of `Anthropic`, the upstream path of the `Anthropic` for `llm/v1/chat` route type is changed from `/v1/complete` to `/v1/messages`"
```
Comment (fffonion) [inline changelog/unreleased/kong-ee/add-messages-api-to-anthropic.yml:1]: would this be a breaking change to mention in changelog?

**Review by fffonion — APPROVED** [review]

File: changelog/unreleased/kong-ee/add-messages-api-to-anthropic.yml (hunk 1 above)
Comment (liverpool8056) [reply changelog/unreleased/kong-ee/add-messages-api-to-anthropic.yml:1]: @fffonion Let me mention it in changelog, thanks!

**Comment by tysoekong** [comment]
Yes this is fantastic, if possible please merge so I can then add the "Streaming" transformation over the top for my PR.

**Comment by tysoekong** [comment]
@liverpool8056

**Review by tysoekong — APPROVED** [review]
Yes this is great, I need to add streaming support too so if any bug is found I will just roll up the fixes into the next commit.

File: kong/llm/drivers/anthropic.lua [hunk 2]
```diff
@@ -96,36 +121,50 @@ local transformers_from = {
...
     local response_table, err = cjson.decode(response_string)
     if err then
-      return nil, "failed to decode cohere response"
+      return nil, "failed to decode anthropic response"
     end
 
-    if response_table.completion then
+    local function extract_text_from_content(content)
+      local buf = buffer:new()
+      buf:reset()
```
Comment (brentos) [inline kong/llm/drivers/anthropic.lua:129]: Do we need to reset the buffer? it was created in the previous line

File: kong/llm/drivers/anthropic.lua [hunk 3]
```diff
@@ -96,36 +121,50 @@ local transformers_from = {
   ["llm/v1/chat"] = function(response_string)
     local response_table, err = cjson.decode(response_string)
     if err then
-      return nil, "failed to decode cohere response"
+      return nil, "failed to decode anthropic response"
     end
 
-    if response_table.completion then
+    local function extract_text_from_content(content)
+      local buf = buffer:new()
```
Comment (brentos) [inline kong/llm/drivers/anthropic.lua:128]:
```suggestion
      local buf = buffer.new()
```

**Review by brentos — COMMENTED** [review]
Just a couple things, otherwise LGTM

**Comment by tysoekong** [comment]
https://github.com/Kong/kong/pull/12792 depends on this too. I can finish off the comments here today.

**Review by chronolaw — APPROVED** [review]

**Pull request merged**